### PR TITLE
Add a way to skip a test

### PR DESCRIPTION
**Pull request by xiazhvera**
*Issue #, if available:*

*Description of changes:*
The test case can be skipped if it returned AWS_OP_SKIP. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: include/aws/testing/aws_test_harness.h [hunk 1]
```diff
@@ -62,9 +62,24 @@ static int s_cunit_failure_message0(
     s_cunit_failure_message0(FAIL_PREFIX, func, file, line, format, #__VA_ARGS__)
 
 static int total_failures;
+static int total_skip;
```
Comment (xiazhvera) [inline include/aws/testing/aws_test_harness.h:65]: The value reserved to record the total skipped tests count. Current not in use.

File: include/aws/common/error.h [hunk 2]
```diff
@@ -14,6 +14,7 @@
 
 #define AWS_OP_SUCCESS (0)
 #define AWS_OP_ERR (-1)
+#define AWS_OP_SKIP (1)
```
Comment (DmitriyMusatkin) [inline include/aws/common/error.h:17]: probably not the best place to define a test only return code. can we put it in some test related header?

File: include/aws/testing/aws_test_harness.h [hunk 3]
```diff
@@ -434,7 +449,7 @@ static inline int s_aws_run_test_case(struct aws_test_harness *harness) {
         test_res |= harness->on_after(allocator, setup_res, harness->ctx);
     }
 
-    if (test_res != AWS_OP_SUCCESS) {
+    if (test_res == AWS_OP_ERR) {
```
Comment (DmitriyMusatkin) [inline include/aws/testing/aws_test_harness.h:452]: should we define anything outside success and skip as error? its corner case, but if test returns something weird outside the defined constants we should probably fail

File: cmake/AwsTestHarness.cmake [hunk 4]
```diff
@@ -87,6 +90,7 @@ function(generate_cpp_test_driver driver_exe_name)
 
     foreach(name IN LISTS TEST_CASES)
         add_test(${name} ${driver_exe_name} "${name}")
+        set_tests_properties("${name}" PROPERTIES SKIP_RETURN_CODE ${SKIP_RETURN_CODE_VALUE})
```
Comment (DmitriyMusatkin) [inline cmake/AwsTestHarness.cmake:93]: do we need this if harness is checking return code of the test point?

Comment (xiazhvera) [reply cmake/AwsTestHarness.cmake:93]: Yes. The ctest still need to know which error code is defined as "skipped" from the harness result.

File: include/aws/common/error.h (hunk 2 above)
Comment (xiazhvera) [reply include/aws/common/error.h:17]: I can move the AWS_OP_SKIP into aws_test_harness.h. However, `AWS_OP_SUCCESS` and `AWS_OP_ERR ` has been used in all the current tests, it makes more sense to have them together. 
Ideally, we should define test specified codes for succes/err/skip in aws_test_harness.h. However, that will be a significant API change that affects all aws-c tests, I dont know if we want to do it now?

Comment (DmitriyMusatkin) [reply include/aws/common/error.h:17]: AWS_OP_SUCCESS and AWS_OP_ERR are also used extensively outside of tests. My concern is we are adding a test specific constant into a common header. In my opinion we should not be polluting this header with test only stuff and putting just the skip macro in test harness is ok as compromise, but we can have a team wide discussion on this.

File: include/aws/testing/aws_test_harness.h [hunk 5]
```diff
@@ -62,9 +67,24 @@ static int s_cunit_failure_message0(
     s_cunit_failure_message0(FAIL_PREFIX, func, file, line, format, #__VA_ARGS__)
 
 static int total_failures;
+static int total_skip;
 
 #define SUCCESS (0)
 #define FAILURE (-1)
+#define SKIP (1)
```
Comment (xiazhvera) [inline include/aws/testing/aws_test_harness.h:74]: The skip code returned to cmake has to be in range [0,255]. Refer to cmake doc: https://cmake.org/cmake/help/latest/prop_test/SKIP_RETURN_CODE.html

**Review by DmitriyMusatkin — APPROVED** [review]
